tokio/tests/sync_mpsc_weak.rs

```diff
@@ -511,3 +512,161 @@ fn test_tx_count_weak_unbounded_sender() {
 
     assert!(tx_weak.upgrade().is_none() && tx_weak2.upgrade().is_none());
 }
+
+#[tokio::test]
```

Comment:
Regarding tests, I personally like the one feature per one test style. In this case, we could split into these tests:

* strong_count is incremented when the Sender is cloned
* weak_count is incremented when the Sender is downgraded
* strong_count is decremented when the Sender is dropped
* weak_count is decremented when the WeakSender is dropped
* mixed cases not covered by the above, if needed

Reply:
That's a good point